server/container_status.go

```diff
@@ -50,26 +51,31 @@ func (s *Server) ContainerStatus(ctx context.Context, req *pb.ContainerStatusReq
 	cState := s.runtime.ContainerStatus(c)
 	rStatus := pb.ContainerState_CONTAINER_UNKNOWN
 
-	switch cState.Status {
-	case oci.ContainerStateCreated:
-		rStatus = pb.ContainerState_CONTAINER_CREATED
-		created := cState.Created.UnixNano()
-		resp.Status.CreatedAt = created
-	case oci.ContainerStateRunning:
-		rStatus = pb.ContainerState_CONTAINER_RUNNING
-		created := cState.Created.UnixNano()
-		resp.Status.CreatedAt = created
-		started := cState.Started.UnixNano()
-		resp.Status.StartedAt = started
-	case oci.ContainerStateStopped:
-		rStatus = pb.ContainerState_CONTAINER_EXITED
-		created := cState.Created.UnixNano()
-		resp.Status.CreatedAt = created
-		started := cState.Started.UnixNano()
-		resp.Status.StartedAt = started
-		finished := cState.Finished.UnixNano()
-		resp.Status.FinishedAt = finished
-		resp.Status.ExitCode = cState.ExitCode
+	if cState != nil {
+		switch cState.Status {
+		case oci.ContainerStateCreated:
+			rStatus = pb.ContainerState_CONTAINER_CREATED
+			created := cState.Created.UnixNano()
+			resp.Status.CreatedAt = created
+		case oci.ContainerStateRunning:
+			rStatus = pb.ContainerState_CONTAINER_RUNNING
+			created := cState.Created.UnixNano()
+			resp.Status.CreatedAt = created
+			started := cState.Started.UnixNano()
+			resp.Status.StartedAt = started
+		case oci.ContainerStateStopped:
+			rStatus = pb.ContainerState_CONTAINER_EXITED
+			created := cState.Created.UnixNano()
+			resp.Status.CreatedAt = created
+			started := cState.Started.UnixNano()
+			resp.Status.StartedAt = started
+			finished := cState.Finished.UnixNano()
+			resp.Status.FinishedAt = finished
+			resp.Status.ExitCode = cState.ExitCode
+		default:
+			resp.Status.CreatedAt = time.Time{}.UnixNano()
+			resp.Status.StartedAt = time.Time{}.UnixNano()
+		}
```

Comment:
Sorry @runcom, saw your "hold off" comment after I'd made it through the whole thing.